### DISCUSSION TITLE
Welcome to istSOS4 Discussions!

### DISCUSSION THREAD

**Discussion by massimiliano-cannata** [istSOS/istSOS4#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by code-walker-23** [comment]
Hello istSOS Community!

I’m Vinay, and I’m excited to be here! I have experience with React, Next.js, TypeScript, and JavaScript, and I’m eager to contribute to the istSOS project. Looking forward to collaborating with everyone and learning more about how I can help!

Best regards,
Vinay

**Comment by MostafaMagdyy** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.


Hello ISTSOS Team,
I am Mostafa Magdy from Egypt, a fourth-year Computer Engineering student at Cairo University with real-world software development experience through internships at startups like [Obelion](https://www.linkedin.com/company/obelion-ai/). I am proficient in C/C++, Python, and JavaScript, and I have completed many projects using these languages. Last year, I made open-source contributions to The Plone Foundation. I am particularly interested in the OSGEO organization and its projects, especially ISTSOS. I have set up the project locally and have tried interfacing with the SensorThings API. However, I am unsure of the next steps.

I haven't found a list of beginner-friendly issues to start with. Should I contribute to the existing issues, or do they require more advanced knowledge?

I am particularly interested in the idea of a Grafana Data Source Plugin for ISTSOS4, as I have experience integrating Grafana to monitor the system and manage logs during my previous internship.

Could you advise on the best way to begin understanding the ISTSOS4 codebase before starting to write a draft proposal for this idea?

**Comment by jph6366** [comment]
Hey istSOS team!

I am geospatial software engineer from Wilmington, North Carolina, USA. I am looking forward to contributing to the project where I can!

In my time working on a proof-of-concept for the project proposals I came across some neat projects that work interoperably with SensorThings API, so I thought I'd share them!

- Internet of Water provides [HubKit](https://github.com/internetofwater/HubKit),  GUI to map their data in .csv files to a SensorThings API data model
- CoatyJS's [Sensor Things objects](https://coatyio.github.io/coaty-js/man/sensor-things-guide/) , framework provides a production-ready application and communication layer foundation for building collaborative IoT applications in an easy-to-use yet powerful and efficient way. 
- [STAM](https://github.com/DataCoveEU/STAM), JavaScript library for showing the Things/Features of interest of a SensorThings server on a Leaflet/OpenLayers map

**Reply by jph6366** [reply]
Exciting collaboration!

[3D and GeoSpatially - Anchored Collaboration on the Web with OGC GeoPose at 3D on the Web 2025
](https://youtu.be/F_TjgtvWpyA?si=BXqDk4JdS8dI_iYm)

![image](https://github.com/user-attachments/assets/ccb355fc-11f1-415f-992c-47d4d3a0497f)
